### ISSUE TITLE
`update --all` not idempotent

### ISSUE THREAD

**Issue by pitrou** [conda/conda#3298]
```
c:\>conda update --all
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment C:\Miniconda3:

The following packages will be UPDATED:

    conda-build: 1.21.7-py35_0              --> 1.21.11-py35_0

The following packages will be SUPERCEDED by a higher-priority channel:

    conda:       4.2.1-py35_0  conda-canary --> 4.1.11-py35_0

Proceed ([y]/n)?

Unlinking packages ...
[      COMPLETE      ]|##################################################| 100%
Linking packages ...
[      COMPLETE      ]|##################################################| 100%

c:\>conda update --all
Using Anaconda API: https://api.anaconda.org
Fetching package metadata ...............
Solving package specifications: ..........

Package plan for installation in environment C:\Miniconda3:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.2.3                |           py35_0         416 KB  conda-canary

The following packages will be UPDATED:

    conda:     4.1.11-py35_0 --> 4.2.3-py35_0 conda-canary
    conda-env: 2.5.2-py35_0  --> 2.6.0-0      conda-canary

Proceed ([y]/n)?

Fetching packages ...
conda-4.2.3-py 100% |###############################| Time: 0:00:00 648.85 kB/s
Extracting packages ...
[      COMPLETE      ]|##################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|##################################################| 100%
Linking packages ...
[      COMPLETE      ]|##################################################| 100%

c:\>conda update --all
Fetching package metadata ...............
Solving package specifications: ..........

# All requested packages already installed.
# packages in environment at C:\Miniconda3:
```

**Comment by HugoTian** [comment]
I cannot reproduce this case, and can you give me the output of conda info and the condarc file ?

Thanks

**Comment by pitrou** [comment]
I am sorry I will not be able to answer. conda-env 2.6.0 is dysfunctional here so I downgraded again and also removed the `conda-canary` channel from my condarc.

**Comment by kalefranz** [comment]
fixed with #3306
